src/roslibpy/comm/comm_cli.py

```diff
@@ -328,6 +331,65 @@ def call_in_thread(self, callback):
         """
         Task.Factory.StartNew(callback, self.cancellation_token)
 
+    def blocking_call_from_thread(self, callback, timeout):
+        """Call the given function from a thread, and wait for the result synchronously
+        for as long as the timeout will allow.
+
+        Args:
+            callback: Callable function to be invoked from the thread.
+            timeout (:obj: int): Number of seconds to wait for the response before
+                raising an exception.
+
+        Returns:
+            The results from the callback, or a timeout exception.
+        """
+        auto_event = AutoResetEvent(False)
```

Comment:
Unless there's I reason I don't see, I would use a `ManualResetEventSlim` here, which works almost identical to the `AutoResetEvent` but we don't need it to reset after waiting, since it's disposed after one usage.

Reply:
Nope, there's no reason you don't see.  Just didn't know about `ManualResetEventSlim`.